### PR TITLE
Aws auth profiles

### PR DESCRIPTION
**Pull request by arturrez**

### CODE REVIEW

File: cmd/nodecmd/create_aws.go [hunk 1]
```diff
@@ -46,27 +46,27 @@ func getNewKeyPairName(ec2Svc *ec2.EC2) (string, error) {
 	}
 }
 
-func printNoCredentialsOutput() {
+func printNoCredentialsOutput(awsProfile string) {
 	ux.Logger.PrintToUser("No AWS credentials file found in ~/.aws/credentials")
```
Comment (felipemadero) [inline cmd/nodecmd/create_aws.go:50]: No AWS credentials found. Either set AWS environment variables ENV_VAR_NAMES_HERE or set config file ~/.aws/credentials.
Improve this printed msg to be clear there are two options (env vars or credential file, what to do with either of those, and where to found more info)

File: cmd/nodecmd/create_aws.go [hunk 2]
```diff
@@ -46,27 +46,27 @@ func getNewKeyPairName(ec2Svc *ec2.EC2) (string, error) {
...
-	ux.Logger.PrintToUser("[default]\naws_access_key_id=<AWS_ACCESS_KEY>\naws_secret_access_key=<AWS_SECRET_ACCESS_KEY>")
+	ux.Logger.PrintToUser("[%s]\naws_access_key_id=<AWS_ACCESS_KEY>\naws_secret_access_key=<AWS_SECRET_ACCESS_KEY>", awsProfile)
 	ux.Logger.PrintToUser("===========END OF FILE===========")
 	ux.Logger.PrintToUser("More info can be found at https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html#file-format-creds")
 }
 
-func printExpiredCredentialsOutput() {
+func printExpiredCredentialsOutput(awsProfile string) {
 	ux.Logger.PrintToUser("AWS credentials expired")
 	ux.Logger.PrintToUser("Fill in ~/.aws/credentials with updated contents following the format below")
```
Comment (felipemadero) [inline cmd/nodecmd/create_aws.go:60]: Either set AWS environment variables ENV_VAR_NAMES_HERE or set config file ~/.aws/credentials.
Improve this printed msg to be clear there are two options (env vars or credential file, what to do with either of those, and where to found more info)

File: cmd/nodecmd/create_aws.go [hunk 3]
```diff
@@ -126,15 +130,15 @@ func getAWSCloudConfig(region string, authorizeAccess bool) (*ec2.EC2, string, s
 			}
 		}
 	}
-	sess, err := getAWSCloudCredentials(region, constants.CreateAWSNode, authorizeAccess)
+	sess, err := getAWSCloudCredentials(awsProfile, region, constants.CreateAWSNode, authorizeAccess)
```
Comment (felipemadero) [inline cmd/nodecmd/create_aws.go:133]: can the parent of this also receive awsProfile as param? just try to minimize global var usage for flags

File: cmd/nodecmd/create_aws.go [hunk 4]
```diff
@@ -152,7 +156,7 @@ func createEC2Instances(rootBody *hclwrite.Body,
 	keyPairName,
 	securityGroupName string,
 ) ([]string, []string, string, string, error) {
-	if err := terraformaws.SetCloudCredentials(rootBody, region); err != nil {
+	if err := terraformaws.SetCloudCredentials(rootBody, awsProfile, region); err != nil {
```
Comment (felipemadero) [inline cmd/nodecmd/create_aws.go:159]: can the parent of this also receive awsProfile as param? just try to minimize global var usage for flags

File: cmd/nodecmd/validatePrimary.go [hunk 5]
```diff
@@ -38,6 +38,7 @@ var (
 	keyName                      string
 	useLedger                    bool
 	useStaticIP                  bool
+	awsProfile                   string
```
Comment (felipemadero) [inline cmd/nodecmd/validatePrimary.go:41]: let's put the var in node create

File: pkg/terraform/aws/terraform_aws.go [hunk 6]
```diff
@@ -20,11 +20,13 @@ import (
 )
 
 // SetCloudCredentials sets AWS account credentials defined in .aws dir in user home dir
-func SetCloudCredentials(rootBody *hclwrite.Body, region string) error {
+func SetCloudCredentials(rootBody *hclwrite.Body, awsProfile, region string) error {
 	provider := rootBody.AppendNewBlock("provider", []string{"aws"})
 	providerBody := provider.Body()
 	providerBody.SetAttributeValue("region", cty.StringVal(region))
-	providerBody.SetAttributeValue("profile", cty.StringVal("default"))
```
Comment (felipemadero) [inline pkg/terraform/aws/terraform_aws.go:27]: not needed anymore if default profile?

File: pkg/terraform/aws/terraform_aws.go [hunk 7]
```diff
@@ -20,11 +20,13 @@ import (
...
+	if awsProfile != constants.AWSDefaultCredential {
```
Comment (felipemadero) [inline pkg/terraform/aws/terraform_aws.go:27]: I mean, why the if?

File: pkg/terraform/aws/terraform_aws.go [hunk 8]
```diff
@@ -210,16 +212,18 @@ func SetOutput(rootBody *hclwrite.Body, useEIP bool) {
 // RunTerraform executes terraform apply function that creates the EC2 instances based on the .tf file provided
 // returns a list of AWS node-IDs and node IPs
 func RunTerraform(terraformDir string, useEIP bool) ([]string, []string, error) {
+	var stdBuffer bytes.Buffer
+	var stderr bytes.Buffer
+	mw := io.MultiWriter(os.Stdout, &stdBuffer)
```
Comment (felipemadero) [inline pkg/terraform/aws/terraform_aws.go:217]: may you try to use SetupRealtimeCLIOutput ?

File: pkg/terraform/aws/terraform_aws.go [hunk 9]
```diff
@@ -210,16 +212,18 @@ func SetOutput(rootBody *hclwrite.Body, useEIP bool) {
...
 	cmd := exec.Command(constants.Terraform, "init") //nolint:gosec
 	cmd.Dir = terraformDir
+	cmd.Stdout = mw
 	if err := cmd.Run(); err != nil {
 		return nil, nil, err
 	}
 	cmd = exec.Command(constants.Terraform, "apply", "-auto-approve") //nolint:gosec
+	cmd.Env = append(cmd.Env, os.Environ()...)
```
Comment (felipemadero) [inline pkg/terraform/aws/terraform_aws.go:225]: why the append?

File: pkg/terraform/terraform.go [hunk 10]
```diff
@@ -50,6 +50,7 @@ func CheckIsInstalled() error {
 
 func GetPublicIPs(terraformDir string) ([]string, error) {
 	cmd := exec.Command(constants.Terraform, "output", "instance_ips") //nolint:gosec
+	cmd.Env = append(cmd.Env, os.Environ()...)
```
Comment (felipemadero) [inline pkg/terraform/terraform.go:53]: why is it needed env here?

File: cmd/nodecmd/create_aws.go (hunk 1 above)
Comment (felipemadero) [reply cmd/nodecmd/create_aws.go:50]: Also be very clear that, in the file,  it is looking for the profile awsProfile. So as to close https://github.com/ava-labs/avalanche-cli/issues/986

File: cmd/nodecmd/create_aws.go (hunk 2 above)
Comment (felipemadero) [reply cmd/nodecmd/create_aws.go:60]: Also be very clear that, in the file,  it is looking for the profile awsProfile. So as to close https://github.com/ava-labs/avalanche-cli/issues/986

**Review by felipemadero — COMMENTED** [review]
Overall LGTM

File: cmd/nodecmd/create_aws.go (hunk 1 above)
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:50]: done

File: cmd/nodecmd/create_aws.go (hunk 3 above)
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:133]: done

File: cmd/nodecmd/create_aws.go (hunk 2 above)
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:60]: done

File: cmd/nodecmd/validatePrimary.go (hunk 5 above)
Comment (arturrez) [reply cmd/nodecmd/validatePrimary.go:41]: wdym?

File: pkg/terraform/aws/terraform_aws.go (hunk 7 above)
Comment (arturrez) [reply pkg/terraform/aws/terraform_aws.go:27]: fixed

File: pkg/terraform/aws/terraform_aws.go (hunk 9 above)
Comment (arturrez) [reply pkg/terraform/aws/terraform_aws.go:225]: sure

File: pkg/terraform/aws/terraform_aws.go (hunk 8 above)
Comment (arturrez) [reply pkg/terraform/aws/terraform_aws.go:217]: I would leave it unchanged as I don't think this PR touches this area

File: pkg/terraform/terraform.go (hunk 10 above)
Comment (arturrez) [reply pkg/terraform/terraform.go:53]: we can just assign it, agree

File: pkg/terraform/aws/terraform_aws.go (hunk 7 above)
Comment (arturrez) [reply pkg/terraform/aws/terraform_aws.go:27]: actually we need it. What if user sets env variables so profile is not needed for terraform.

File: cmd/nodecmd/create_aws.go [hunk 11]
```diff
@@ -46,27 +46,32 @@ func getNewKeyPairName(ec2Svc *ec2.EC2) (string, error) {
 	}
 }
 
-func printNoCredentialsOutput() {
-	ux.Logger.PrintToUser("No AWS credentials file found in ~/.aws/credentials")
-	ux.Logger.PrintToUser("Create a file called 'credentials' with the contents below, and add the file to ~/.aws/ directory")
+func printNoCredentialsOutput(awsProfile string) {
+	ux.Logger.PrintToUser("No AWS credentials found in file ~/.aws/credentials ")
+	ux.Logger.PrintToUser("or in env variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY")
```
Comment (felipemadero) [inline cmd/nodecmd/create_aws.go:51]: Or in env ...

File: cmd/nodecmd/create_aws.go [hunk 12]
```diff
@@ -46,27 +46,32 @@ func getNewKeyPairName(ec2Svc *ec2.EC2) (string, error) {
...
 	ux.Logger.PrintToUser("More info can be found at https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html#file-format-creds")
+	ux.Logger.PrintToUser("Also you can set environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY")
+	ux.Logger.PrintToUser("Please use https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-set for more details")
+
 }
 
-func printExpiredCredentialsOutput() {
+func printExpiredCredentialsOutput(awsProfile string) {
 	ux.Logger.PrintToUser("AWS credentials expired")
 	ux.Logger.PrintToUser("Fill in ~/.aws/credentials with updated contents following the format below")
```
Comment (felipemadero) [inline cmd/nodecmd/create_aws.go:65]: also tell the user about env variables and profile name to use in file here

File: cmd/nodecmd/create.go [hunk 13]
```diff
@@ -136,7 +139,7 @@ func createNodes(_ *cobra.Command, args []string) error {
 	gcpCredentialFilepath := ""
 	if cloudService == constants.AWSCloudService {
 		// Get AWS Credential, region and AMI
-		ec2Svc, region, ami, err := getAWSCloudConfig(cmdLineRegion, authorizeAccess)
+		ec2Svc, region, ami, err := getAWSCloudConfig(awsProfile, cmdLineRegion, authorizeAccess)
```
Comment (felipemadero) [inline cmd/nodecmd/create.go:142]: also make createAWSInstances to receive awsProfile, so there is no more global var assumption on create_aws.go

**Review by felipemadero — COMMENTED** [review]
after this comments, and lint etc, LGTM will approve

File: cmd/nodecmd/create_aws.go (hunk 12 above)
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:65]: done

File: cmd/nodecmd/create.go (hunk 13 above)
Comment (arturrez) [reply cmd/nodecmd/create.go:161]: done

**Review by felipemadero — APPROVED** [review]

**Pull request merged**